build_modules/lib/src/kernel_builder.dart

```diff
@@ -76,39 +77,47 @@ Future _createKernel(
     @required bool summaryOnly,
     @required String outputExtension,
     @required String sdkKernelPath}) async {
-  var transitiveDeps = await module.computeTransitiveDependencies(buildStep);
-  var transitiveKernelDeps = <AssetId>[];
-  var transitiveSourceDeps = <AssetId>[];
-
-  // Provide kernel dependencies where possible (if created in a previous
-  // phase), otherwise provide dart sources.
-  await Future.wait(transitiveDeps.map((dep) async {
-    var kernelId = dep.primarySource.changeExtension(outputExtension);
-    if (await buildStep.canRead(kernelId)) {
-      transitiveKernelDeps.add(kernelId);
-    } else {
-      transitiveSourceDeps.addAll(dep.sources);
-    }
-  }));
+  var request = new WorkRequest();
+  AssetId outputId;
+  File outputFile;
+  File packagesFile;
```

Comment:
why were these moved up?

Reply:
They are used below but assigned inside the block so that the temporary sets within the block can be gc'd

Reply:
Ah, I missed that 84 to 152 is now a block

I think we need to split this out into some submethods since it's nearly 100 lines and the structure is getting hard to follow.

Reply:
done